Source/JavaScriptCore/wasm/WasmIPIntGenerator.cpp

```diff
@@ -594,23 +594,41 @@ PartialResult WARN_UNUSED_RETURN IPIntGenerator::getLocal(uint32_t index, Expres
 }
 PartialResult WARN_UNUSED_RETURN IPIntGenerator::setLocal(uint32_t index, ExpressionType)
 {
-    auto len = m_parser->offset() - m_parser->currentOpcodeStartingOffset() - 1;
+    auto len = m_parser->offset() - m_parser->currentOpcodeStartingOffset();
     if (index >= m_metadata->m_numArguments)
         m_metadata->addLEB128ConstantInt32AndLength(index + m_metadata->m_nonArgLocalOffset, len);
     else
         m_metadata->addLEB128ConstantInt32AndLength(m_metadata->m_argumentLocations[index], len);
     return { };
 }
 
-PartialResult WARN_UNUSED_RETURN IPIntGenerator::getGlobal(uint32_t, ExpressionType&) { return { }; }
-PartialResult WARN_UNUSED_RETURN IPIntGenerator::setGlobal(uint32_t, ExpressionType) { return { }; }
+PartialResult WARN_UNUSED_RETURN IPIntGenerator::getGlobal(uint32_t index, ExpressionType&)
+{
+    auto len = m_parser->offset() - m_parser->currentOpcodeStartingOffset();
```

Comment:
We seem to repeat this a lot

Reply:
Yup, added a helper function in IPIntGenerator to fetch this value. Also updated constants to remove the + 1 (and just handle that in the metadata generator).